src/main/java/io/kestra/plugin/kestra/executions/Search.java

```diff
@@ -0,0 +1,225 @@
+package io.kestra.plugin.kestra.executions;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.common.FetchOutput;
+import io.kestra.core.models.tasks.common.FetchType;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.plugin.kestra.AbstractKestraTask;
+import io.kestra.sdk.KestraClient;
+import io.kestra.sdk.internal.ApiException;
+import io.kestra.sdk.model.*;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.annotation.Nullable;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import reactor.core.publisher.Flux;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@SuperBuilder(toBuilder = true)
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Search for Kestra Executions"
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Search for execution with specific label>",
+            full = true,
+            code = """
+                id: search_executions_by_label
+                namespace: company.team.myflow
+                
+                tasks:
+                  - id: hello
+                    type: io.kestra.plugin.kestra.executions.Search
+                    kestraUrl: http://localhost:8080
+                    labels:
+                      key: value
+                    auth:
+                      username: root@root.com
+                      password: Root!1234
+                    fetchType: STORE # Store the results in a file
+                """
+        ),
+        @Example(
+            title = "Search for execution in success",
+            full = true,
+            code = """
+                id: search_success_executions
+                namespace: company.team.myflow
+                
+                tasks:
+                  - id: search_executions
+                    type: io.kestra.plugin.kestra.executions.Search
+                    kestraUrl: http://localhost:8080
+                    timeRange: PT10H # In the last 10 hours
+                    state:
```

Comment:
```suggestion
                    states:
```

since it accepts a list of states, let's change it to plural, agree?